pkg/governance-scripts/contracts/20230109-gauge-adder-migration-v2-to-v3/GaugeAdderMigrationCoordinator.sol

```diff
@@ -0,0 +1,142 @@
+
+    constructor(
+        IAuthorizerAdaptor authorizerAdaptor,
+        IGaugeAdder _newGaugeAdder,
+        IGaugeAdder _oldGaugeAdder,
+        ILiquidityGaugeFactory _ethereumGaugeFactory,
+        ILiquidityGaugeFactory _polygonRootGaugeFactory,
+        ILiquidityGaugeFactory _arbitrumRootGaugeFactory,
+        ILiquidityGaugeFactory _optimismRootGaugeFactory,
+        address _liquidityMiningCommitteeMultisig,
+        address _gaugeCheckpointingMultisig
+    ) BaseCoordinator(authorizerAdaptor) {
+        newGaugeAdder = _newGaugeAdder;
+        oldGaugeAdder = _oldGaugeAdder;
+        ethereumGaugeFactory = _ethereumGaugeFactory;
+        polygonRootGaugeFactory = _polygonRootGaugeFactory;
+        arbitrumRootGaugeFactory = _arbitrumRootGaugeFactory;
+        optimismRootGaugeFactory = _optimismRootGaugeFactory;
+        liquidityMiningCommitteeMultisig = _liquidityMiningCommitteeMultisig;
+        gaugeCheckpointingMultisig = _gaugeCheckpointingMultisig;
+
+        IGaugeController gaugeControllerAddress = _newGaugeAdder.getGaugeController();
+        require(
+            gaugeControllerAddress == _oldGaugeAdder.getGaugeController(),
+            "Gauge controller doesn't match on adders"
+        );
+        gaugeController = gaugeControllerAddress;
+    }
+
+    // Coordinator Setup
+
+    function _registerStages() internal override {
+        _registerStage(_firstStage);
+    }
+
+    function _firstStage() private {
+        _setupNewGaugeAdder();
+        _deprecateOldGaugeAdder();
+    }
+
+    function _afterLastStage() internal virtual override {
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        authorizer.renounceRole(authorizer.DEFAULT_ADMIN_ROLE(), address(this));
+    }
+
+    // Internal functions
+
+    function _setupNewGaugeAdder() private {
+        ICurrentAuthorizer authorizer = ICurrentAuthorizer(address(getAuthorizer()));
+
+        // Set up factories on new gauge adder.
+        // We don't pull these from the old gauge adder as this uses uncapped gauges,
+        // whereas we want to add capped gauges to the new adder.
+        {
+            bytes32 addFactoryRole = IAuthentication(address(newGaugeAdder)).getActionId(
+                IGaugeAdder.addGaugeFactory.selector
+            );
+            authorizer.grantRole(addFactoryRole, address(this));
+
+            newGaugeAdder.addGaugeFactory(ethereumGaugeFactory, IGaugeAdder.GaugeType.Ethereum);
+            newGaugeAdder.addGaugeFactory(polygonRootGaugeFactory, IGaugeAdder.GaugeType.Polygon);
+            newGaugeAdder.addGaugeFactory(arbitrumRootGaugeFactory, IGaugeAdder.GaugeType.Arbitrum);
+            newGaugeAdder.addGaugeFactory(optimismRootGaugeFactory, IGaugeAdder.GaugeType.Optimism);
+
+            authorizer.renounceRole(addFactoryRole, address(this));
+        }
+
+        // Grant permissions for adding new Gauges.
+        // Permissions for Gnosis and ZKSync are not granted.
```

Comment:
Maybe say why we're holding back on the new chains (e.g., because the factories aren't deployed). Also mention BSC?